includes/wikia/parser/templatetypes/handlers/InfoIconTemplate.class.php

```diff
@@ -33,19 +37,24 @@ public static function handle( $wikitext, Parser $parser ) {
 	 * @return String
 	 * @throws \MWException
 	 */
-	private static function makeIconLink( $image ) {
-		$link = '';
-		$title = Title::newFromText( $image );
-		$file = wfFindFile( $title );
-		if ( $file && $file->exists() ) {
-			$link = Linker::makeImageLink2(
-				$title,
-				$file,
-				[],
-				[ 'template-type' => TemplateClassificationService::TEMPLATE_INFOICON ]
-			);
+	private static function getIconLink($image ) {
+		// SRE-117: Use process cache to avoid repeated expensive media queries
+		if ( !isset( self::$infoIconCache[$image] ) ) {
+			$title = Title::newFromText( $image );
```

Comment:
while at it - Title newFromText can return null, should we guard against it here?

Reply:
hmm, good point. I could add a check for safety, although I don't recall seeing errors from here.